lxd/instance_post.go

```diff
@@ -87,84 +90,71 @@ func instancePost(d *Daemon, r *http.Request) response.Response {
 	// Flag indicating whether the node running the container is offline.
 	sourceNodeOffline := false
 
-	// Flag indicating whether the node the container should be moved to is
-	// online (only relevant if "?target=<node>" was given).
-	targetNodeOffline := false
-
-	// A POST to /containers/<name>?target=<node> is meant to be used to
-	// move a container from one node to another within a cluster.
-	if targetNode != "" {
-		// Determine if either the source node (the one currently
-		// running the container) or the target node are offline.
-		//
-		// If the target node is offline, we return an error.
-		//
-		// If the source node is offline and the container is backed by
-		// ceph, we'll just assume that the container is not running
-		// and it's safe to move it.
-		//
-		// TODO: add some sort of "force" flag to the API, to signal
-		//       that the user really wants to move the container even
-		//       if we can't know for sure that it's indeed not
-		//       running?
-		err := s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
-			p, err := dbCluster.GetProject(ctx, tx.Tx(), projectName)
-			if err != nil {
-				return fmt.Errorf("Failed loading project: %w", err)
-			}
-
-			apiProject, err := p.ToAPI(ctx, tx.Tx())
-			if err != nil {
-				return err
-			}
-
-			// Check if user is allowed to use cluster member targeting
-			err = project.CheckClusterTargetRestriction(r, apiProject, targetNode)
-			if err != nil {
-				return err
-			}
-
-			// Load target node.
-			node, err := tx.GetNodeByName(ctx, targetNode)
-			if err != nil {
-				return fmt.Errorf("Failed to get target node: %w", err)
-			}
-
-			targetNodeOffline = node.IsOffline(s.GlobalConfig.OfflineThreshold())
+	// Check if clustered.
+	clustered, err := cluster.Enabled(s.DB.Node)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("Failed checking cluster state: %w", err))
+	}
 
-			// Load source node.
-			address, err := tx.GetNodeAddressOfInstance(ctx, projectName, name, instanceType)
-			if err != nil {
-				return fmt.Errorf("Failed to get address of instance's member: %w", err)
-			}
+	var inst instance.Instance
+	var targetProject *api.Project
+	var targetMemberInfo *db.NodeInfo
+	var candidateMembers []db.NodeInfo
 
-			if address == "" {
-				// Local node.
-				sourceNodeOffline = false
-				return nil
-			}
+	// Load requested instance
+	inst, err = instance.LoadByProjectAndName(s, projectName, name)
+	if err != nil {
+		return response.SmartError(err)
+	}
 
-			node, err = tx.GetNodeByAddress(ctx, address)
-			if err != nil {
-				return fmt.Errorf("Failed to get source member for %s: %w", address, err)
-			}
+	target := queryParam(r, "target")
+	if !clustered && target != "" {
+		return response.BadRequest(fmt.Errorf("Target only allowed when clustered"))
+	}
 
-			sourceNodeOffline = node.IsOffline(s.GlobalConfig.OfflineThreshold())
+	// A POST to /containers/<name>?target=<node> is meant to be used to
```

Comment:
As we're moving this comment can we update it to use "instances" concept rather than "containers" to bring it inline with todays terminology. Also the API route is /instances/ now.

Reply:
Also, we use the term "cluster member" or just "member" (if the context is inherently cluster related) in all user-facing messages/documentation rather than "node". I've found it helps for consistency to also use cluster member terminology in the code/comments too.

Reply:
By keeping an eye on both scenarios clustered an non clustered, I have updated the comments to address the wording.